submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,197 @@
+## Globalnet 2.0: Enhancements to Globalnet Implementation
+
+## Summary
+
+Currently, Submariner Globalnet implementation in [release 0.8](https://github.com/submariner-io/releases/releases/tag/v0.8.0)
+annotates every Pod, Service and Node in the cluster with a globalIp to support Pod to remote
+Service connectivity in the following manner.
+
+Once a Pod is scheduled and enters Running state, it is annotated with a globalIp. To support
+the connectivity, Globalnet programs the necessary egress rules on the active Gateway Node of
+the cluster where the source IP of the Pod is modified to the corresponding globalIp. This
+translation enables us to uniquely identify the traffic from the Pod, when it reaches the
+destination cluster.
+
+Similarly, a ClusterIP Service is annotated with a globalIp so that remote Pods can communicate
+with it using its globalIp. Globalnet programs the necessary ingress rules on the active Gateway
+node of the cluster, where the service resides. The ingress rules enable any incoming traffic
+destined to the globalIp of the Service to be redirected to the corresponding ClusterIP Service.
+
+In order to support connectivity from the Host Network to the remote services, Globalnet also
+annotates every node in the cluster with a unique globalIp.
+
+While this works fine for small clusters, where the number of Pods and Services are limited, for
+large clusters it poses scalability and performance issues as we would eventually run out of
+globalIps because of annotating every Pod/Service/Node object. Also, in this design, Pod to Pod
+connectivity as well as Headless Services are not supported.
+
+## Proposal
+
+This enhancement proposes the following changes to make Globalnet implementation scale better.
+
+1. By default, every cluster will be assigned a single globalIp which will be used as egress IP
+   for any cross-cluster communication.
+2. Provision to allocate a globalIp at the namespace/project level which takes precedence over
+   the globalIp allocated to the cluster.
+3. Provision to allocate a globalIp to a Pod (or a set of Pods) which has higher precedence
+   over globalIPs at namespace level. If there are multiple policies, users can specify a priority
+   field for each of the policy.
+4. Instead of annotating every ClusterIP Service in the cluster, only exported Services will be
+   annotated with a globalIp.
```

Comment:
This is great and I feel like we can start doing it regardless of the other proposals straight away, kind of "globalnet 1.5"

Reply:
I dont know if we can backport this to 0.8.2 (or something) as its a feature and user-experience will be different. But this can surely go as part of the initial 0.9 release if other Globalnet 2.0 changes take some additional time.

Reply:
I think it's better not to spend more time in the existing globalnet, and go ahead using our time with v2

Reply:
But since we will not be backporting it to 0.8 release. Lets keep it part of Globalnet 2.0 work.